save-backend/src/main/kotlin/com/saveourtool/save/backend/service/LnkContestProjectService.kt

```diff
@@ -65,4 +69,37 @@ class LnkContestProjectService(
         lnkContestProjectRepository.save(LnkContestProject(project, contest, null, 0.0))
         true
     }
+
+    /**
+     * @param newExecution
+     */
+    @Transactional
+    fun updateBestExecution(newExecution: Execution) {
+        val newScore = requireNotNull(newExecution.score) {
+            "Cannot update best score, because no score has been provided for execution id=${newExecution.id}"
+        }
+        val contest = lnkContestExecutionService.findContestByExecution(newExecution)
+            ?: error("Execution was performed not as a part of contest")
+        val project = newExecution.project
+        val lnkContestProject = lnkContestProjectRepository.findByContestAndProject(contest, project)
+            .orElseThrow {
+                IllegalStateException("Project ${project.shortToString()} is not bound to contest name=${contest.name}")
+            }
+        val oldBestScore = lnkContestProject.bestScore
+        if (oldBestScore == null || oldBestScore <= newScore) {
+            logger.debug {
+                "For project ${project.shortToString()} updating best_score from execution " +
+                        "[id=${lnkContestProject.bestExecution?.id},score=$oldBestScore] to " +
+                        "[id=${newExecution.id},score=$newScore]"
+            }
+            lnkContestProject.bestExecution = newExecution
+            lnkContestProject.bestScore = newExecution.score
```

Comment:
do we need `bestScore` as separate field on `lnkContestProject` now? it can be replaced to `bestExecution?.score`